src/System.Runtime.Analyzers/Core/InstantiateArgumentExceptionsCorrectly.cs

```diff
@@ -1,59 +1,189 @@
         internal static DiagnosticDescriptor IncorrectMessageRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageIncorrectMessage,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: s_helpUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
         internal static DiagnosticDescriptor IncorrectParameterNameRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageIncorrectParameterName,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: s_helpUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(NoArgumentsRule, IncorrectMessageRule, IncorrectParameterNameRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(
+                compilationContext =>
+                {
+                    Compilation compilation = compilationContext.Compilation;
+                    ITypeSymbol stringType = compilation.GetSpecialType(SpecialType.System_String);
```

Comment:
You don't need to fetch/store the special types. Instead you can directly ask for the special type on the type symbols. Basically replace your `argument.Parameter.Type != stringType` checks with `argument.Parameter.Type.SpecialType != SpecialType.System_String`